components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UserStoreActionListener.java

```diff
@@ -46,15 +46,15 @@ public int getExecutionOrderId() {
     @Override
     public boolean doPreAddUser(String userName, Object credential, String[] roleList, Map<String, String> claims,
                                 String profile, UserStoreManager userStoreManager) throws UserStoreException {
+
         if (!isEnable()) {
             return true;
         }
-
         try {
+            AddUserWFRequestHandler addUserWFRequestHandler = new AddUserWFRequestHandler();
```

Comment:
Is there any specific reason to move this to here? it seems previous implementation don't have any issue